timetable/extractor.py

```diff
@@ -63,56 +63,47 @@ def create_timings(_table: Tag | NavigableString) -> list[int]:
 
     timings = create_timings(table)
     rows = table.find_all('tr')
+    
     for row in rows:
         day_cell: PageElement = row.find('td', {'valign': 'top'})
         if not day_cell:
             continue
-        day = days[day_cell.get_text()] if day_cell.get_text() in days.keys() else day_cell.get_text()
-
-        # Merge timeslots occurring adjacent to each other and initialize course objects
+        day = days.get(day_cell.get_text(), day_cell.get_text())
+    
         prev: Course | None = None
         course_duration: int = 0
```

Comment:
This code was removed iirc. Is your branch up to date?

Reply:
Ah no it wasn't up to date. End of day bug is solved then.

Reply:
If your improvements still simplify the code, please update the branch, and I'll merge it.

Reply:
I have updated the branch. Please check once.